### PR TITLE
(#548) Try to fix linker issues again

### PR DESCRIPTION
**Pull request by herrhotzenplotz**
mostly using the wrong library (GNU Scientific Library instead of the correct Microsoft.GSL one)

### CODE REVIEW

**Comment by herrhotzenplotz** [comment]
I hope this fixes the linker and dependency confusions once and forever :smile:

Disabling the dependency to termbench when not building the tests is hopefully fine. Maybe that could be solved differently, but not within the scope of this PR I guess.

File: scripts/install-deps.sh [hunk 1]
```diff
@@ -201,8 +204,8 @@ install_deps_FreeBSD()
         qt5-qmake \
         qt5-widgets \
         range-v3 \
+        yaml-cpp
```
Comment (christianparpart) [inline scripts/install-deps.sh:207]: Oh you have gotten that to work? That would be awesome. I prefer system deps whenever possible.

File: scripts/install-deps.sh [hunk 2]
```diff
@@ -216,6 +219,7 @@ install_deps_arch()
         fmt \
         fontconfig \
         git \
+        gsl \
```
Comment (christianparpart) [inline scripts/install-deps.sh:222]: Many thanks for also taking care of other platforms. - A safe exit would be to always embed a dependency, but that should be last resort.

File: src/crispy/CMakeLists.txt [hunk 3]
```diff
@@ -31,7 +31,7 @@ set(crispy_SOURCES
 add_library(crispy-core ${crispy_SOURCES})
 add_library(crispy::core ALIAS crispy-core)
 
-set(CRISPY_CORE_LIBS range-v3::range-v3 fmt::fmt-header-only unicode::core GSL)
+set(CRISPY_CORE_LIBS range-v3::range-v3 fmt::fmt-header-only unicode::core Microsoft.GSL::GSL)
```
Comment (christianparpart) [inline src/crispy/CMakeLists.txt:34]: And this works for all platforms? I hope CI will give us a :+1: here :)

**Review by christianparpart — APPROVED** [review]
> I hope this fixes the linker and dependency confusions once and forever 😄

I am praying for that one now, too and hope that CI will now pick up FreeBSD issues as early as possible, too.

> Disabling the dependency to termbench when not building the tests is hopefully fine. Maybe that could be solved differently, but not within the scope of this PR I guess.

Oh that is absolutely fine. I personally try to keep PRs as atomic (topic-wise) as possible, but I know very good that this is not always feasible. Keeping it practical sometimes wins. ;)

**Comment by christianparpart** [comment]
FYI: the failing external tests is due to AddressSanatizer picking up a memory leak on libfontconfig. I have a fix for that that at least works locally on Ubuntu 21.10, I seem to need to fix either more memory leaks based on the Ubuntu 20.04 provided fontconfig or simply `fetch` the latest stable and patch that (I want to submit a patch for that fix somewhat soonish, too, so I don't have to keep patching other libs just to get CI green :) )

File: src/crispy/CMakeLists.txt (hunk 3 above)
Comment (herrhotzenplotz) [reply src/crispy/CMakeLists.txt:34]: As far as I can see, this is the way how you're supposed to do it: https://github.com/Microsoft/GSL#usage-in-cmake

It built on all the platforms that the CI tests for :man_shrugging: I guess it is fine then.

**Pull request merged**